src/interfaces/assistants_web/src/components/Agents/AgentsList.tsx

```diff
@@ -1,44 +1,113 @@
 'use client';
 
-import { Transition } from '@headlessui/react';
+import { useMemo, useState } from 'react';
 
+import { ConversationWithoutMessages as Conversation } from '@/cohere-client';
 import { AgentCard } from '@/components/Agents/AgentCard';
-import { Text } from '@/components/Shared';
+import { ConversationListLoading } from '@/components/ConversationList/ConversationListLoading';
+import { ConversationListPanelGroup } from '@/components/ConversationList/ConversationListPanelGroup';
+import { Icon, Input, Text } from '@/components/Shared';
 import { useRecentAgents } from '@/hooks/agents';
-import { useAgentsStore } from '@/stores';
+import { useConversations } from '@/hooks/conversation';
+import { useSearchConversations } from '@/hooks/search';
+
+const sortByDate = (a: Conversation, b: Conversation) => {
+  return Date.parse(b.updated_at ?? '') - Date.parse(a.updated_at ?? '');
+};
 
 /**
  * @description This component renders a list of agents.
  * It shows the most recent agents and the base agents.
  */
 export const AgentsList: React.FC = () => {
-  const {
-    agents: { isAgentsSidePanelOpen },
-  } = useAgentsStore();
   const { recentAgents } = useRecentAgents();
+  const { data: conversations, isLoading: isConversationsLoading, isError } = useConversations({});
+  const { search, setSearch, searchResults } = useSearchConversations(conversations);
+  const hasSearchQuery = search.length > 0;
+  const hasSearchResults = searchResults.length > 0;
+  const hasConversations = conversations.length > 0;
+  const [checkedConversations, setCheckedConversations] = useState<Set<string>>(new Set());
```

Comment:
nit: can we reorganize the order?

maybe keep the hooks together and then the variables separated by a new line